edge-modules/MetricsCollector/Program.cs

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace MetricsCollector
+{
+    using System;
+    using System.Linq;
+    using System.Runtime.Loader;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Client;
+    using Microsoft.Azure.Devices.Client.Transport.Mqtt;
+    using Microsoft.Azure.Devices.Edge.ModuleUtil;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Azure.Devices.Shared;
+    using Microsoft.Extensions.Logging;
+    using Newtonsoft.Json;
+
+    internal class Program
+    {
+        static readonly Version ExpectedSchemaVersion = new Version("1.0");
+        static readonly ILogger Logger = ModuleUtil.CreateLogger("MetricsCollector");
+        static Timer scrapingTimer;
+
+        public static int Main() => MainAsync().Result;
+
+        private static async Task<int> MainAsync()
+        {
+            Logger.LogInformation($"Starting metrics collector with the following settings:\r\n{Settings.Current}");
+
+            await Init();
+
+            (CancellationTokenSource cts, ManualResetEventSlim completed, Option<object> handler) = ShutdownHandler.Init(TimeSpan.FromSeconds(5), Logger);
+
+            // Wait until the app unloads or is cancelled
+            AssemblyLoadContext.Default.Unloading += ctx => cts.Cancel();
+            Console.CancelKeyPress += (sender, cpe) => cts.Cancel();
+            await WhenCancelled(cts.Token, completed, handler);

```

Comment:
```suggestion
            await WhenCancelled(cts.Token, completed, handler).ConfigureAwait(false);
```

Reply:
We shouldn't need this since metrics collector is running on dotnet core, right?

Reply:
Adding as it should still be compatible with .NET standard

Reply:
https://stackoverflow.com/questions/56349117/configureawait-in-net-core-console-worker-not-asp-net-core

I think you're right Andrew.